table/block_fetcher.cc

```diff
@@ -360,15 +359,15 @@ IOStatus BlockFetcher::ReadBlockContents() {
       return io_status_;
     }
   } else if (!TryGetSerializedBlockFromPersistentCache()) {
-    ReadBlock(/*retry =*/false, fs_buf);
+    ReadBlock(/*retry =*/false, fs_buf_);
```

Comment:
nit: Since `fs_buf_` is now a member of `block_fetcher`, I think  `ReadBlock()` no longer needs to take `FSAllocationPtr&` as a param and just do `fs_buf_ = std::move(read_req.fs_scratch);`?